### PR TITLE
Switch to setup-terraform github action

### PR DESCRIPTION
**Pull request by aidy**
terraform-github-actions deprecated

connects #262

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,21 +20,17 @@ jobs:
...
-      - name: 'Terraform Format'
-        uses: hashicorp/terraform-github-actions@master
+      - uses: actions/checkout@main
+      - uses: hashicorp/setup-terraform@v1
         with:
-          tf_actions_version: 0.12.24
-          tf_actions_subcommand: 'fmt'
+          terraform_version: 0.12.24
+      - name: 'Terraform Format'
+        run: terraform fmt -check
```
Comment (errm) [inline .github/workflows/ci.yml:28]: Might want to add -recursive here

Comment (aidy) [reply .github/workflows/ci.yml:28]: Ah, yes - good call.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -20,23 +20,15 @@ jobs:
...
-      - name: 'Terraform Format'
-        uses: hashicorp/terraform-github-actions@master
+      - uses: actions/checkout@main
+      - uses: hashicorp/setup-terraform@v1
         with:
-          tf_actions_version: 0.12.24
-          tf_actions_subcommand: 'fmt'
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          terraform_version: 0.12.24
```
Comment (ettiee) [inline .github/workflows/ci.yml:26]: Spotted that this doesn't line up with https://github.com/cookpad/terraform-aws-eks/blob/main/.terraform-version
Could we improve this by reading that file to an environment variable to use in `terraform_version` ? Would be nice to only define it in 1 place in the project

Comment (aidy) [reply .github/workflows/ci.yml:26]: Yes - we should do something to have them sync.

Not here though - keeping it as 0.12.24 so it's an atomic change.

**Pull request merged**